### PR TITLE
Feature/sxssf partial implementation

### PR DESCRIPTION
**Pull request by rstpv**
Trying to enable SXSSF with spoiwo I used most of the code  from Model2XslxConversions as SXSSF is mostly a wrapper around XSSF
Things that don't work:

- Writing or accessing  existing cells from a template
- createTable is not available in SXSSFSheet
- Odd and Even headers and footers are not available in SXSSFSheet 

Most of the tests are from xlsx using xssf, just changed imports and type definitions to validate that things still worked.

As there was too many duplicated code (I started with a full copy of Model2XlsxConversions), I created a base trait with shared functionality, not sure if this is the recommended way to go.

Can you please give me some feedback?

Thank you.

### CODE REVIEW

**Comment by norbert-radyk** [comment]
I am afraid I will be only able to look into your change after 7th May.
Apologies for the delay.

File: src/main/scala/com/norbitltd/spoiwo/natures/xlsx/Model2XlsxConversions.scala [hunk 1]
```diff
@@ -320,91 +159,19 @@ object Model2XlsxConversions {
...
-    sp.printGridLines.foreach(sheet.setPrintGridlines)
-    sp.rightToLeft.foreach(sheet.setRightToLeft)
-    sp.rowSumsBelow.foreach(sheet.setRowSumsBelow)
-    sp.rowSumsRight.foreach(sheet.setRowSumsRight)
-    sp.selected.foreach(sheet.setSelected)
-    sp.tabColor.foreach(c => sheet.setTabColor(convertColor(c)))
-    sp.virtuallyCenter.foreach(sheet.setVerticallyCenter)
-    sp.zoom.foreach(sheet.setZoom)
+  override def setTabColor(sheet: usermodel.Sheet, color: XSSFColor) {
+    sheet.asInstanceOf[SXSSFSheet].setTabColor(color)
```
Comment (norbert-radyk) [inline src/main/scala/com/norbitltd/spoiwo/natures/xlsx/Model2XlsxConversions.scala]: This should use a non-streaming sheet.

Comment (rstpv) [reply src/main/scala/com/norbitltd/spoiwo/natures/xlsx/Model2XlsxConversions.scala]: fixed

**Comment by rstpv** [comment]
What would you think that would be the best aproach to avoid the full internal model in memory?
Do you think it would be a good idea to use a streams solution like Akka streams or better to stick to scala native Streams + foreach?

**Comment by norbert-radyk** [comment]
Can you please elaborate a bit how you'd imagine using Akka Streams for the report generation? 

While both SXSSF and Akka Streams have streaming in the name, to me they address slightly different category of problems. I see Akka as a great solution for handling the back-pressure when dealing with large or infinite streams of data. This is hardly the case when generating Excel spreadsheet where the number of rows is limited to 1M. 

SXSSF is a great solution to avoid storing this one million row in memory before being put in a file. I would imagine that is kind of what would be required from SPOIWO as well - give an ability to process one row at a time, rather than all rows. That should be achievable with the standard Scala libraries. 

By the way, now I am happy to merge your change as is and we can iterate on the more memory efficient model generation part. Are you happy for me to merge?

**Comment by rstpv** [comment]
I was thinking about how using streams in order to reduce the memory footprint, making for example the sheets/rows not all evaluated at once but to be accessed as they go, and then write them to the SXSSF workbook and, as they are being processed as they go the memory of each one would be free to be Garbage collected (I still hadn't figured a way to do so).

If you think this is good enough as a first aproach, I would be happy for you to merge it.

**Pull request merged**